docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,90 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workload running on top
+of it.
+
+MetalK8s provides monitoring service with Prometheus and AlertManager in
+order to expose system performance time series and alerts.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information that can be very helpful when it comes to understand the
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to collect and aggregate all logs in order to
+make them browsable from a central endpoint. It is also important to ease
+browsing and correlation with other type of indicators such as KPIs and alerts.
+
+In addition to the application logs, the platform and workload are also
+producing some audit logs. If the same system is used to manage application and
+audit logs, then it is important to provide the end users of the workloads
+some capabilities to secure and query the audit logs belonging to their usage.
+
+Requirements
+------------
+
+- Collect both containers and system logs (STDOUT or STDERR)
+- Persistent storage of logs
+- Horizontally scalable (capacity and performance)
+- Accessible from a central end point
+- Highly Available
+- Easy to use query language
+- Service accessible through a UI
+- Monitorable service (health, performances and alerts)
+- Ability to correlate logs with KPIs and alerts
+- Multitenancy for audit logs
+- Performances (TBD)
+
+
+User Stories
+------------
+
+- As a Platform Administrator, I want to browse all MetalK8s containers logs
+  (from all servers) from a unique endpoint, in order to ease distributed
+  system error investigation
+
+- As a Platform Administrator, I want to browse all system logs (from all
```

Comment:
Again, what are system logs? How do I group logs? Per service? Per server?

Reply:
Here I am talking about the System so maybe I should say Platform? I don't know how we should group logs ... you mean how we should label logs? to ease the browsing?

Reply:
I think we should stick to a single definition of "system logs", as opposed to "container logs". Since we can manage them easily by name, let's only consider "systemd-managed services" under "system logs" for now. The exceptions (like `/var/log/metalk8s`) will need to be treated as exceptions anyway.

Yes, for browsing / dashboards that we would have to provide (we don't have any today, maybe should be in the requirements? also extending the existing dashboards with some log panels?)

Reply:
so now we are talking about journal logs.
About Grafana dashboards, what is important is the ability to display logs in Grafana. With time and experience, we will then build predefined dashboards (but can't specify it now)